fairlearn/reductions/_moments/conditional_selection_rate.py

```diff
@@ -67,7 +157,26 @@ def gamma(self, predictor):
 
     # TODO: this can be further improved using the overcompleteness in group membership
     def project_lambda(self, lambda_vec):
-        """Return the projected lambda values."""
+        r"""Return the rebalanced Lagrange multipliers.
+
+        For each constraint, we have two Lagrange multipliers corresponding to the
+        positive and negative violations of the constraint, with both multipliers
+        constrained to be positive. The 'true' Lagrange multiplier is then
+        :math:`\lambda = \lambda_{+} - \lambda_{-}`. Since it makes little
+        sense to the positive and negative constraints violated at the same time,
+        this routine rebalances between them. For example, if we have
+        :math:`\lambda_{+}=3` and :math:`\lambda_{-}=1` then this method will
+        make :math:`\lambda_{+}=2` and :math:`\lambda_{-}=0`. Similarly if
+        :math:`\lambda_{+}=2` and :math:`\lambda_{-}=3` then this method will
+        yield :math:`\lambda_{+}=0` and :math:`\lambda_{-}=1`.
+
+        The supplied :class:`pandas:pandas.DataFrame` must have the same
```

Comment:
Is this constraint on `index` correct? It would seem to make sense, since otherwise this method would never reference `self`. If so, I should add the check.

Reply:
this explanation is too operational. the right way is to go from the moment-level definition of project_lambda.

Reply:
I've simplified the text somewhat, although I'm not quite sure what you mean by the "moment-level definition of `project_lambda`. I've had a look at the OneNote, and this `project_lambda` doesn't return `argmax` of anything - although it's only used by `_eval_from_error_gamma` in the `_Lagrangian` and `max()` values are taken there. Also, this rebalancing between the positive and negative multipliers is going to be necessary to discover the true largest $\lambda$